[package] sdl/2.x: impossible to build with VS 2026

### Description

I guess this is the reason - CMake < 4 doesn't know about VS 2026:
https://github.com/conan-io/conan-center-index/blob/20431e49ce487414b4d39fb55eb299b1e32e98c2/recipes/sdl/all/conanfile.py#L185

Possible local workarounds would be `tools.cmake.cmaketoolchain:generator=Ninja` or `platform_tool_requires` + `replace_tool_requires` for locally installed CMake.

P.S. Upgrading to SDL v3 isn't an option for me right now.

### Package and Environment Details

* Package Name/Version: **sdl/2.32.10**
* Operating System+version: **Windows 11**
* Compiler+version: **MSVC v145**
* Conan version: **conan 2.29**
* Python version: **Python 3.14.4**


### Conan profile

```
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=195
os=Windows
glaze/*:compiler.cppstd=23
&:build_type=Debug

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=195
os=Windows
glaze/*:compiler.cppstd=23
```

### Steps to reproduce

conan install --requires sdl/2.32.10

### Logs

<details><summary>Click to expand log</summary>

```
sdl/2.32.10: Calling source() in C:\Users\kambala\.conan2\p\sdl37ccb0f66d55e\s\src
sdl/2.32.10: Sources correctly downloaded from https://www.libsdl.org/release/SDL2-2.32.10.tar.gz
sdl/2.32.10: Uncompressing SDL2-2.32.10.tar.gz to .
sdl/2.32.10: Apply patch (file): patches/2.32.10-0001-alsa-fix-signature-of-snd_pcm_info_free.patch

-------- Installing package sdl/2.32.10 (56 of 80) --------
sdl/2.32.10: Building from source
sdl/2.32.10: Package sdl/2.32.10:ca5cda8d512510f89f9b053c70f0ef211ea8a32d
sdl/2.32.10: settings: os=Windows arch=x86_64 compiler=msvc compiler.runtime=dynamic compiler.runtime_type=Release compiler.version=195 build_type=Release
sdl/2.32.10: options: directx=True hidapi=True opengl=True opengles=True sdl2main=True shared=True vulkan=True
sdl/2.32.10: Copying sources to build folder
sdl/2.32.10: Building your package in C:\Users\kambala\.conan2\p\b\sdl9caea44575780\b
sdl/2.32.10: Writing generators to C:\Users\kambala\.conan2\p\b\sdl9caea44575780\b\build\generators
sdl/2.32.10: Generator 'CMakeDeps' calling 'generate()'
sdl/2.32.10: Generator 'PkgConfigDeps' calling 'generate()'
sdl/2.32.10: Generator 'VirtualBuildEnv' calling 'generate()'
sdl/2.32.10: Calling generate()
sdl/2.32.10: Generators folder: C:\Users\kambala\.conan2\p\b\sdl9caea44575780\b\build\generators
sdl/2.32.10: CMakeToolchain generated: conan_toolchain.cmake
sdl/2.32.10: CMakeToolchain generated: C:\Users\kambala\.conan2\p\b\sdl9caea44575780\b\build\generators\CMakePresets.json
sdl/2.32.10: CMakeToolchain generated: C:\Users\kambala\.conan2\p\b\sdl9caea44575780\b\src\CMakeUserPresets.json
sdl/2.32.10: Generating aggregated env files
sdl/2.32.10: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
sdl/2.32.10: Calling build()
sdl/2.32.10: Running CMake.configure()
sdl/2.32.10: RUN: cmake -G "Visual Studio 18 2026" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/kambala/.conan2/p/b/sdl9caea44575780/p" -DCMAKE_POLICY_DEFAULT_CMP0077="NEW" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:/Users/kambala/.conan2/p/b/sdl9caea44575780/b/src"
CMake Error: Could not create named generator Visual Studio 18 2026

Generators
  Visual Studio 17 2022        = Generates Visual Studio 2022 project files.
                                 Use -A option to specify architecture.
  Visual Studio 16 2019        = Generates Visual Studio 2019 project files.
                                 Use -A option to specify architecture.
  Visual Studio 15 2017 [arch] = Generates Visual Studio 2017 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 14 2015 [arch] = Generates Visual Studio 2015 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Borland Makefiles            = Generates Borland makefiles.
* NMake Makefiles              = Generates NMake makefiles.
  NMake Makefiles JOM          = Generates JOM makefiles.
  MSYS Makefiles               = Generates MSYS makefiles.
  MinGW Makefiles              = Generates a make file for use with
                                 mingw32-make.
  Green Hills MULTI            = Generates Green Hills MULTI files
                                 (experimental, work-in-progress).
  Unix Makefiles               = Generates standard UNIX makefiles.
  Ninja                        = Generates build.ninja files.
  Ninja Multi-Config           = Generates build-<Config>.ninja files.
  Watcom WMake                 = Generates Watcom WMake makefiles.
  CodeBlocks - MinGW Makefiles = Generates CodeBlocks project files
                                 (deprecated).
  CodeBlocks - NMake Makefiles = Generates CodeBlocks project files
                                 (deprecated).
  CodeBlocks - NMake Makefiles JOM
                               = Generates CodeBlocks project files
                                 (deprecated).
  CodeBlocks - Ninja           = Generates CodeBlocks project files
                                 (deprecated).
  CodeBlocks - Unix Makefiles  = Generates CodeBlocks project files
                                 (deprecated).
  CodeLite - MinGW Makefiles   = Generates CodeLite project files
                                 (deprecated).
  CodeLite - NMake Makefiles   = Generates CodeLite project files
                                 (deprecated).
  CodeLite - Ninja             = Generates CodeLite project files
                                 (deprecated).
  CodeLite - Unix Makefiles    = Generates CodeLite project files
                                 (deprecated).
  Eclipse CDT4 - NMake Makefiles
                               = Generates Eclipse CDT 4.0 project files
                                 (deprecated).
  Eclipse CDT4 - MinGW Makefiles
                               = Generates Eclipse CDT 4.0 project files
                                 (deprecated).
  Eclipse CDT4 - Ninja         = Generates Eclipse CDT 4.0 project files
                                 (deprecated).
  Eclipse CDT4 - Unix Makefiles= Generates Eclipse CDT 4.0 project files
                                 (deprecated).
  Kate - MinGW Makefiles       = Generates Kate project files (deprecated).
  Kate - NMake Makefiles       = Generates Kate project files (deprecated).
  Kate - Ninja                 = Generates Kate project files (deprecated).
  Kate - Ninja Multi-Config    = Generates Kate project files (deprecated).
  Kate - Unix Makefiles        = Generates Kate project files (deprecated).
  Sublime Text 2 - MinGW Makefiles
                               = Generates Sublime Text 2 project files
                                 (deprecated).
  Sublime Text 2 - NMake Makefiles
                               = Generates Sublime Text 2 project files
                                 (deprecated).
  Sublime Text 2 - Ninja       = Generates Sublime Text 2 project files
                                 (deprecated).
  Sublime Text 2 - Unix Makefiles
                               = Generates Sublime Text 2 project files
                                 (deprecated).


sdl/2.32.10: ERROR:
Package 'ca5cda8d512510f89f9b053c70f0ef211ea8a32d' build failed
sdl/2.32.10: WARN: Build folder C:\Users\kambala\.conan2\p\b\sdl9caea44575780\b\build
ERROR: sdl/2.32.10: Error in build() method, line 318
        cmake.configure()
        ConanException: Error 1 while executing
```

</details>

--- a/recipes/sdl/3.x/conanfile.py
+++ b/recipes/sdl/3.x/conanfile.py
@@ -236,7 +236,7 @@ class SDLConan(ConanFile):
             self.requires("xorg/system")
 
     def build_requirements(self):
-        self.tool_requires("cmake/[>=3.24 <4]")
+        self.tool_requires("cmake/[>=3.24]")
         if self._is_unix_sys and not self.conf.get("tools.gnu:pkg_config", check_type=str):
             self.tool_requires("pkgconf/[>=2.2 <3]")
         if self.options.get_safe("wayland"):

--- a/recipes/sdl/all/conanfile.py
+++ b/recipes/sdl/all/conanfile.py
@@ -151,10 +151,11 @@ class SDLConan(ConanFile):
         if self.options.get_safe("iconv", False):
             self.requires("libiconv/1.17")
         if self.settings.os == "Linux":
-            if self.options.alsa:
-                self.requires("libalsa/[>=1.2 <1.3]")
             if self.options.pulse:
                 self.requires("pulseaudio/14.2")
+            if self.options.alsa:
+                # Version comes from pulseaudio
+                self.requires("libalsa/[>=1.2 <1.3]")
             if self.options.opengl:
                 self.requires("opengl/system")
             if self.options.nas:

--- a/recipes/sdl/all/conanfile.py
+++ b/recipes/sdl/all/conanfile.py
@@ -152,7 +152,7 @@ class SDLConan(ConanFile):
             self.requires("libiconv/1.17")
         if self.settings.os == "Linux":
             if self.options.alsa:
-                self.requires("libalsa/1.2.10")
+                self.requires("libalsa/[>=1.2 <1.3]")
             if self.options.pulse:
                 self.requires("pulseaudio/14.2")
             if self.options.opengl:
@@ -160,8 +160,9 @@ class SDLConan(ConanFile):
             if self.options.nas:
                 self.requires("nas/1.9.5")
             if self.options.wayland:
-                self.requires("wayland/1.22.0")
                 self.requires("xkbcommon/1.6.0")
+                # Version comes from xkbcommon
+                self.requires("wayland/[^1.22]")
                 self.requires("egl/system")
             if self.options.libunwind:
                 self.requires("libunwind/1.8.0")
@@ -182,11 +183,11 @@ class SDLConan(ConanFile):
                 raise ConanInvalidConfiguration("Package for 'directfb' is not available (yet)")
 
     def build_requirements(self):
-        self.tool_requires("cmake/[>3.27 <4]")
+        self.tool_requires("cmake/[>=3.24]")
         if self.settings.os == "Linux" and not self.conf.get("tools.gnu:pkg_config", check_type=str):
-            self.tool_requires("pkgconf/2.1.0")
+            self.tool_requires("pkgconf/[>=2.1.0 <3]")
         if hasattr(self, "settings_build") and self.options.get_safe("wayland"):
-            self.build_requires("wayland/1.22.0")  # Provides wayland-scanner
+            self.build_requires("wayland/<host_version>")  # Provides wayland-scanner
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)

